### ISSUE TITLE
Stopped working on HA 2023.7

### ISSUE THREAD

**Issue by LordShadowen** [patrickhilker/tedee_hass_integration#26]
After updating, the integration page shows this:

![image](https://github.com/patrickhilker/tedee_hass_integration/assets/10828042/9f7e3311-968f-4ac3-bbc2-008c46772330)

The logs shows this:

```
2023-07-05 23:35:20.012 ERROR (MainThread) [custom_components.tedee.coordinator] 'NoneType' object is not subscriptable
2023-07-05 23:35:20.013 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Tedee' for tedee integration not ready yet: ('Querying API failed. Error: %s', TypeError("'NoneType' object is not subscriptable")); Retrying in background
```

**Comment by zweckj** [comment]
Weird. Everything’s working flawlessly for me… Can you enable debug logging?

**Comment by dancameo2k** [comment]
I have the same issue after updating Tedee Integration from 2022 to 2023 version:

2023-07-06 21:48:41.549 ERROR (MainThread) [custom_components.tedee.coordinator] 'NoneType' object is not subscriptable
2023-07-06 21:48:51.721 ERROR (MainThread) [custom_components.tedee.coordinator] 'NoneType' object is not subscriptable
2023-07-06 21:48:51.722 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Tedee' for tedee integration not ready yet: ('Querying API failed. Error: %s', TypeError("'NoneType' object is not subscriptable")); Retrying in background
2023-07-06 21:48:57.509 ERROR (MainThread) [custom_components.tedee.coordinator] 'NoneType' object is not subscriptable
2023-07-06 21:49:06.792 DEBUG (MainThread) [custom_components.tedee.coordinator] Update coordinator: Getting locks from API
2023-07-06 21:49:07.188 ERROR (MainThread) [custom_components.tedee.coordinator] 'NoneType' object is not subscriptable
2023-07-06 21:49:07.189 DEBUG (MainThread) [custom_components.tedee.coordinator] Finished fetching tedee API coordinator data in 0.396 seconds (success: False)
2023-07-06 21:49:07.189 WARNING (MainThread) [homeassistant.config_entries] Config entry 'Tedee' for tedee integration not ready yet: ('Querying API failed. Error: %s', TypeError("'NoneType' object is not subscriptable")); Retrying in background

**Comment by zweckj** [comment]
Could you please test pre-release 2023.7.3b1?

**Comment by LordShadowen** [comment]
2023.7.3b1 works for me

**Comment by zweckj** [comment]
thanks for testing, closing as fixed in [2023.7.3](https://github.com/patrickhilker/tedee_hass_integration/releases/tag/2023.7.3)
